src/OmniSharp.Roslyn/BufferManager.cs

```diff
@@ -128,37 +133,53 @@ private bool TryAddTransientDocument(string fileName, string fileContent)
             var projects = FindProjectsByFileName(fileName);
             if (!projects.Any())
             {
+                if (fileName.EndsWith(".cs") && _workspace.TryAddMiscellaneousDocument(fileName, LanguageNames.CSharp) != null)
```

Comment:
I was going to say something about adding `.csx` here but they have different project rules anyway.

Reply:
I think we should be doing a case-insensitive `EndsWith` here to ensure that we also catch `.CS`.